src/core/core.controller.js

```diff
@@ -529,6 +529,9 @@ module.exports = function(Chart) {
 
 			me.drawDatasets(easingValue);
 
+			// notify before drawing tooltips
+			plugins.notify(me, 'beforeTooltipDraw', [easingValue]);
```

Comment:
What about making this new hook consistent with `*DatasetDraw`:
- make it cancelable: if the hook return `false`, the tooltip should not be drawn
- use the new hook signature that accepts only one argument: `hook(chart, args, options)`

We might want to move that code in a new function, similar to [`drawDataset`](/chartjs/Chart.js/blob/b9afeaf973999516cd3fdfdf4ec7c86b223d9263/src/core/core.controller.js#L580):

```javascript
/**
 * @private
 */
_drawTooltip: function(easingValue) {
    var me = this;
    var tooltip = me.tooltip;
    var args = {
        tooltip: tooltip ,
        easingValue: easingValue
    };

    if (plugins.notify(me, 'beforeTooltipDraw', [args]) === false) {
        return;
    }

    tooltip .draw();

    plugins.notify(me, 'afterTooltipDraw', [args]);
},
```


Then in `draw()`:
```javascript
    // ...
    me.drawDatasets(easingValue);
    me._drawTooltip(easingValue);
    
    plugins.notify(me, 'afterDraw', [easingValue]);
    // ...
```

Thoughts?

Reply:
easingValue doesn't seem to be used when drawing the tooltip... any reason to pass it as an arument to the drawTooltip function anyway?  Also, any particular reason to call it _drawTooltip instead of just drawTooltip?

Reply:
`easingValue` to be consistent with other `*Draw` hooks (it's more or less the animation progress). We can still add it later on demand if you think it's useless right now. `_drawTooltip` because it's a private method (there are many other private methods not prefixed but we switched to that convention too late).